consensus/hotstuff/vote_collector.go

```diff
@@ -23,17 +26,25 @@ const (
 	VoteCollectorStatusInvalid
 )
 
-// VoteCollector collects votes for the same block, produces QC when enough votes are collected
-// VoteCollector takes a callback function to report the event that a QC has been produced.
-type VoteCollector interface {
+func (ps VoteCollectorStatus) String() string {
+	names := [...]string{"VoteCollectorStatusCaching",
+		"VoteCollectorStatusVerifying",
+		"VoteCollectorStatusInvalid"}
+	if ps < VoteCollectorStatusCaching || ps > VoteCollectorStatusInvalid {
+		return "UNKNOWN"
+	}
+	return names[ps]
+}
+
+// VoteCollectorState collects votes for the same block, produces QC when enough votes are collected
+// VoteCollectorState takes a callback function to report the event that a QC has been produced.
+type VoteCollectorState interface {
 	// AddVote adds a vote to the collector
-	// returns true if the vote was added
-	// returns false otherwise
 	// return error if the signature is invalid
 	// When enough votes have been added to produce a QC, the QC will be created asynchronously, and
 	// passed to EventLoop through a callback.
 	// ToDo: document sentinel errors expected during normal operation
-	AddVote(vote *model.Vote) (bool, error)
+	AddVote(vote *model.Vote) error
```

Comment:
I think that returning if what was added or not just adds implementation complexity but doesn't give any useful information to caller.